pkg/config/config.go

```diff
@@ -104,7 +104,9 @@ type Server struct {
 	LogLevel       string `def:"info" desc:"log level: debug|info|warn|error" mapstructure:"log-level"`
 	BadgerLogLevel string `def:"error" desc:"log level: debug|info|warn|error" mapstructure:"badger-log-level"`
 
-	StoragePath     string `def:"<installPrefix>/var/lib/pyroscope" desc:"directory where pyroscope stores profiling data" mapstructure:"storage-path"`
+	StoragePath          string  `def:"<installPrefix>/var/lib/pyroscope" desc:"directory where pyroscope stores profiling data" mapstructure:"storage-path"`
+	StorageDiskThreshold float64 `def:"5" desc:"percentage of available disk space at which ingestion requests are discarded. Defaults to 5%. Set 0 to disable" mapstructure:"storage-disk-threshold"`
```

Comment:
I would call it something a little more explicit. Right now it's unclear from the name what this parameter does.

Maybe `MinFreeSpacePercentage`?